ayaneo-platform.c

```diff
@@ -266,139 +280,71 @@ static int write_ec_ram(u8 index, u8 val)
-                write_ec_ram(zone[zoneindex], state);
-                ayaneo_led_mc_close(CLOSE_CMD_2);
+        for (zone = 0; zone < 4; zone++) {
+                ayaneo_led_mc_set(group, zones[zone], color[0]);
+                ayaneo_led_mc_set(group, zones[zone] + 1, color[1]);
+                ayaneo_led_mc_set(group, zones[zone] + 2, color[2]);
         }
-        ayaneo_led_mc_write();
 }
 
-/* Open every LED register as write enabled */
-static void ayaneo_led_mc_apply(void)
+static void ayaneo_led_mc_off(void)
 {
-        ayaneo_led_mc_state(AYANEO_LED_MC_ON);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x70, 0x0);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xb2, 0xba);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x72, 0xba);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xbf, 0x0);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x7f, 0x0);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xc0, 0x0);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x80, 0x0);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xc1, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x81, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xc2, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x82, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xc3, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x83, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xc4, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x84, 0x5);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0xc5, 0x7);
-        ayaneo_led_mc_close(CLOSE_CMD_2);
-
-        ayaneo_led_mc_open();
-        write_ec_ram(0x85, 0x7);
-        ayaneo_led_mc_close(CLOSE_CMD_1);
-
-        ayaneo_led_mc_write();
+        ec_write_ram(0xb2, 0x31);
+        ec_write_ram(0x86, 0x01);
+        mdelay(AYANEO_LED_WRITE_DELAY_MS);
 }
 
-static void ayaneo_led_mc_color(u8 *color)
+static void ayaneo_led_mc_on(void)
 {
-        u8 quadrant;
-
-        // Zone 1 (Left Stick)
-        ayaneo_led_mc_open();
-        for(quadrant = 0; quadrant < 12; quadrant++) {
-                write_ec_ram(0xB3 + quadrant, color[quadrant % 3]); // Quadrant 1
-        }
-        ayaneo_led_mc_write();
+        ec_write_ram(0xb2, 0xb1);
```

Comment:
These two functions could be combined if we establish a const for 0xb1 and 0xc1

Reply:
Yeah, I think on/off/reset could all be a single function, but I was trying to keep the intent behind each clear since they're used in various places.
I've introduced constants for the addresses and values at least so the writes make sense.